### PR TITLE
kudo: operators structure change and bump to v0.8.0-pre.2

### PR DESCRIPTION
**Pull request by zmalik**
This PR:

- updates the operators structure to use tasks
- bumps to KUDO `v0.8.0-pre.2`
- changes to get the tests ✅with new release of KUDO

### CODE REVIEW

File: repository/zookeeper/operator/templates/validation.yaml
Comment (samvantran) [inline repository/zookeeper/operator/templates/validation.yaml]: changing to hardcoded values intentional?

Comment (gerred) [reply repository/zookeeper/operator/templates/validation.yaml]: this an internal validation process, we may want it to go away entirely.

Comment (samvantran) [reply repository/zookeeper/operator/templates/validation.yaml]: oh I just saw this is a merge of `dev -> master` so the change has already been made

Comment (zmalik) [reply repository/zookeeper/operator/templates/validation.yaml]: it's a validation pod, we might introduce later configurable values for this one. But all it has to do is one call to zookeeper and finish its job. It's not a long-running pod and just used in bootstrap.

The issue was that it was using the same resources `{{ .Params.MEMORY }}` and `{{ .Params.CPUS }}` as the zookeeper nodes when it doesn't have the same responsibilities

Comment (zen-dog) [reply repository/zookeeper/operator/templates/validation.yaml]: > changing to hardcoded values intentional?

💯 This plan is about validating ZK connection and doesn't have to be parameterized (or use the same amount of cpu/mem as ZK itself). Additionally, using the same amount of cpu/mem doesn't work with our CI plan.

> this an internal validation process, we may want it to go away entirely.

I found it useful. Also, this is pretty much the only showcase _right now_ of why bother with extra plans at all 🤷‍♂

**Review by zen-dog — APPROVED** [review]

File: repository/zookeeper/operator/operator.yaml
Comment (zen-dog) [inline repository/zookeeper/operator/operator.yaml:54]: Fun fact: this `delete: true` was a bug: effectively this plan was deleting a non-existing job (`validation.yaml`) and was immediately successful. So no validation happened. This is now fixed on the right side /cc @gerred @zmalik

Comment (gerred) [reply repository/zookeeper/operator/operator.yaml:54]: nice @zen-dog - that's awesome.

**Comment by gerred** [comment]
@zmalik we will have one more breaking change with the apiVersions that @alenkacz is working on, so maybe we should hold this PR open a little longer.

**Comment by zen-dog** [comment]
> @zmalik we will have one more breaking change with the apiVersions that @alenkacz is working on, so maybe we should hold this PR open a little longer.

This would be a new kudoctl build (v0.8.0-pre.2), right? Merging `dev` into `master` will be an ongoing process, so I guess, this PR is meaningful on its own so we could merge this one and open another one when `apiVersions` lands 🤷‍♂

**Comment by zmalik** [comment]
> @zmalik we will have one more breaking change with the apiVersions that @alenkacz is working on, so maybe we should hold this PR open a little longer.

yeah I agree with @zen-dog here that we should merge this PR and open new PRs once some API changes are introduced. As we have tagged releases like `v0.8.0-pre.1` and the next one `v0.8.0-pre.2`

**Comment by zmalik** [comment]
rebased from master as there were conflicts in this PR and also bumped to kudo `0.8.0-pre.2`

**Comment by zmalik** [comment]
rebase broke the tests. Trying to reproduce the test failure in local, and will push a fix ASAP I get it ✅ in local

**Review by alenkacz — APPROVED** [review]
✅

**Pull request merged**
